### PR TITLE
Checkbox: Add extra small checkboxes

### LINKED ISSUE #2991: [Enhancement] Smaller Checkboxes
## Describe the enhancement
Smaller Checkboxes that is 16x16 this might be a bug because the xs options gives a 24x24px so does sm and md. 

## Describe the solution you'd like
Checkboxes that are 16x16px
image

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [ ] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

### PR DESCRIPTION
**Pull request by AgreSanGit**
## Which issue does this PR close?

This PR closes #2991 

## What is the new behavior?

Extra small checkbox now gives a smaller checkbox.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [ ] Determine if your changes are a fix, feature or breaking-change, and add the matching label to your PR. If it is tooling, dependency updates or similar, add ignore-for-release.
- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [ ] Request that the changes are code-reviewed 
- [ ] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

**Review by mark-drastrup — CHANGES_REQUESTED** [review]
Looks really good! My only comment is to update the cookbook text, that says "The checkbox and checkmark always have the same size.", and to update the vertical placement of the "xs: 32px" in the example :)

File: libs/designsystem/checkbox/src/checkbox.component.scss [hunk 1]
```diff
@@ -109,24 +113,35 @@ $default-checkbox-radio-size: map.get(utils.$checkbox-radio-sizes, 'md');
     .wrapper {
       position: relative;
       display: inline-flex;
-      align-items: flex-start;
+      align-items: center;
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/checkbox/src/checkbox.component.scss]: I think this is the right choice, but note that the multiline checkbox looks different now:
Before: 
image

After: 
image

File: apps/cookbook/src/app/showcase/checkbox-showcase/checkbox-showcase.component.html [hunk 2]
```diff
@@ -43,7 +43,7 @@ <h2>States</h2>
   <h2>Sizes</h2>
   <p>
     The size applies to the clickable/tapable area (highlighted in the examples below). The checkbox
-    and checkmark always have the same size.
+    and checkmark have the same size unless you choose XS.
```
Comment (RasmusKjeldgaard) [inline apps/cookbook/src/app/showcase/checkbox-showcase/checkbox-showcase.component.html]: Consider wording here, so the difference is a bit easier to understand. How about something like:  

> The small and medium sizes apply to the clickable/tapable area (highlighted in the examples below) while the checkbox itself remains the same size. 
> 
> The extra small size applies to both click-area and checkbox, but should be used with caution. 
> 
> The extra small size is mainly intended for use in expert systems or information dense interfaces where the reduced size is a necessity but note that it comes at the cost of degraded accessibility.

**Review by RasmusKjeldgaard — CHANGES_REQUESTED** [review]

File: libs/designsystem/checkbox/src/checkbox.component.scss (hunk 1 above)
Comment (RasmusKjeldgaard) [reply libs/designsystem/checkbox/src/checkbox.component.scss]: Verify with UX and I also think we should aim to get this into v9 so we can add a small note about this alignment to the migration notes. Teams might have compensated for this themselves and aligned the checkbox horizontally.

File: apps/cookbook/src/app/showcase/checkbox-showcase/checkbox-showcase.component.html (hunk 2 above)
Comment (AgreSanGit) [reply apps/cookbook/src/app/showcase/checkbox-showcase/checkbox-showcase.component.html]: Check

File: libs/designsystem/checkbox/src/checkbox.component.scss (hunk 1 above)
Comment (RasmusKjeldgaard) [reply libs/designsystem/checkbox/src/checkbox.component.scss]: Scratch that, after discussing with our UI designers it is clear that this should be reverted so the checkbox is top-aligned 😃

File: libs/designsystem/calendar/src/calendar.component.scss [hunk 3]
```diff
@@ -96,6 +96,7 @@ td {
 .day.disabled,
 .day:not(.selectable) {
   --color: #{utils.get-text-color('semi-dark')};
+
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/calendar/src/calendar.component.scss]:
```suggestion
```

File: libs/designsystem/checkbox/src/checkbox.component.scss [hunk 4]
```diff
@@ -125,27 +122,51 @@ $spacing-to-label: map.get(utils.$checkbox-radio-spacing, 'to-label');
...
+      ion-checkbox::part(label) {
+        line-height: utils.line-height('xs');
+        font-size: #{utils.font-size('s')};
+      }
 
-  ion-checkbox {
-    margin: 0;
+      ion-checkbox {
+        --size: #{$checkbox-icon-size-xs};
+        --border-radius: 4px;
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/checkbox/src/checkbox.component.scss]: Consider using utils.sizes here

File: libs/designsystem/checkbox/src/checkbox.component.scss [hunk 5]
```diff
@@ -125,27 +122,51 @@ $spacing-to-label: map.get(utils.$checkbox-radio-spacing, 'to-label');
...
+          padding: utils.size('xxxxs'); // Spacing between checkmark and container box
+        }
+      }
+    }
+
+    ion-checkbox {
+      position: static;
+      flex-shrink: 0;
+      flex-grow: 0;
+    }
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/checkbox/src/checkbox.component.scss]: Consider removing this block - I do not think it has any effect on the layout?

File: apps/cookbook/src/app/showcase/checkbox-showcase/checkbox-showcase.component.html [hunk 6]
```diff
@@ -41,8 +41,16 @@ <h2>States</h2>
...
+<p>
+  The extra small size applies to both click-area and checkbox, but should be used with caution.
+</p>
+<p>
+  The extra small size is mainly intended for use in expert systems or information dense interfaces
+  where the reduced size is a necessity but note that it comes at the cost of degraded
+  accessibility.
 </p>
 <cookbook-example-viewer [html]="checkboxSizesExample.template">
   <cookbook-checkbox-sizes-example #checkboxSizesExample></cookbook-checkbox-sizes-example>
```
Comment (RasmusKjeldgaard) [inline apps/cookbook/src/app/showcase/checkbox-showcase/checkbox-showcase.component.html:56]: Consider adding a divider inside the `cookbook-checkbox-sizes-example` so it looks something like this:

```
<kirby-checkbox size="xs" text="Extra Small"></kirby-checkbox>
<kirby-divider [hasMargin]="true"></kirby-divider>
<kirby-checkbox size="sm" text="Small"></kirby-checkbox>
<kirby-checkbox size="md" text="Medium (default)"></kirby-checkbox>
```

image

File: libs/designsystem/checkbox/src/checkbox.component.scss [hunk 7]
```diff
@@ -125,27 +122,51 @@ $spacing-to-label: map.get(utils.$checkbox-radio-spacing, 'to-label');
...
-    @each $size, $height in utils.$checkbox-radio-sizes {
+    @each $size, $properties in utils.$checkbox-sizes {
+      :host-context(.checkbox-#{$size}),
       &.#{$size} {
         ion-checkbox::part(label) {
-          $vertical-padding: get-vertical-padding($height);
+          $vertical-padding: map.get($properties, vertical-padding);
 
           padding-top: $vertical-padding;
           padding-bottom: $vertical-padding;
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/checkbox/src/checkbox.component.scss]: Consider some light boyscouting 😇 
```suggestion
          padding-block: $vertical-padding;
```

File: libs/designsystem/checkbox/src/checkbox.component.scss [hunk 8]
```diff
@@ -125,27 +122,51 @@ $spacing-to-label: map.get(utils.$checkbox-radio-spacing, 'to-label');
...
         }
       }
     }
-  }
-}
 
-:host-context(kirby-item) {
-  z-index: utils.z('default'); // Makes whole kirby-item clickable above item-inner.
+    :host-context(.checkbox-xs),
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/checkbox/src/checkbox.component.scss]: This should not be necessary, as far as I remember it was an old cookbook thing that leaked into our implementation
```suggestion
```

**Review by RasmusKjeldgaard — CHANGES_REQUESTED** [review]

File: libs/designsystem/checkbox/src/checkbox.component.scss [hunk 9]
```diff
@@ -125,27 +122,51 @@ $spacing-to-label: map.get(utils.$checkbox-radio-spacing, 'to-label');
   &.has-label {
     display: block;
 
-    @each $size, $height in utils.$checkbox-radio-sizes {
+    @each $size, $properties in utils.$checkbox-sizes {
+      :host-context(.checkbox-#{$size}),
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/checkbox/src/checkbox.component.scss]: This should not be necessary, as far as I remember it was an old cookbook thing that leaked into our implementation

```suggestion
```

File: libs/designsystem/checkbox/src/checkbox.component.scss [hunk 10]
```diff
@@ -125,27 +122,42 @@ $spacing-to-label: map.get(utils.$checkbox-radio-spacing, 'to-label');
...
     }
-  }
-}
 
-:host-context(kirby-item) {
-  z-index: utils.z('default'); // Makes whole kirby-item clickable above item-inner.
+    &.xs {
+      ion-checkbox::part(label) {
+        line-height: utils.line-height('xs');
+        font-size: #{utils.font-size('s')};
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/checkbox/src/checkbox.component.scss]:
```suggestion
        font-size: utils.font-size('s');
```

**Review by RasmusKjeldgaard — APPROVED** [review]

**Pull request merged**